airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionHandler.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.integrations.util
+
+import io.airbyte.cdk.integrations.base.AirbyteTraceMessageUtility
+import io.airbyte.cdk.integrations.base.Command
+import io.airbyte.cdk.integrations.base.errors.messages.ErrorMessage
+import io.airbyte.commons.exceptions.ConfigErrorException
+import io.airbyte.commons.exceptions.ConnectionErrorException
+import io.airbyte.commons.exceptions.TransientErrorException
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.function.Consumer
+import kotlin.system.exitProcess
+import org.jetbrains.annotations.VisibleForTesting
+
+private val LOGGER = KotlinLogging.logger {}
+
+const val COMMON_EXCEPTION_MESSAGE_TEMPLATE: String =
+    "Could not connect with provided configuration. Error: %s"
+
+const val DATABASE_CONNECTION_ERROR: String =
+    "Encountered an error while connecting to the database error"
+
+enum class FailureType {
+    CONFIG,
+    TRANSIENT
+}
+
+data class ConnectorErrorProfile(
+    var errorClass: String,
+    var regexMatchingPattern: String,
+    var failureType: FailureType,
+    var externalMessage: String,
+    var sampleInternalMessage: String,
+    var referenceLink: List<String>
+)
```

Comment:
consider having some kind of validation in an `init { ... }` block, for instance to validate that `regexMatchingPattern` is a valid pattern

Reply:
I was not aware of this `init` mechanism; adding it now, thanks!!

Reply:
init block is now added, thanks for the suggestion!

Reply:
kotlin has some nice syntactic sugar :)